ops/op2/signature.rs

```diff
@@ -249,16 +308,33 @@ pub struct ParsedSignature {
   pub generic_bounds: BTreeMap<String, String>,
 }
 
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+enum BufferMode {
+  /// Shared buffers that may possibly change on the JavaScript side upon re-entry into
+  /// V8. Rust code should not treat these as traditional buffers.
+  Shared,
```

Comment:
This is a bit confusing since SharedArrayBuffer is a thing but this doesn't actually refer to that.

Reply:
Correct -- I don't have a better name, bikeshedding appreciated!

Reply:
Hmm... Unsafe / Unsound would be technically correct, and would also serve as a reminder that this isn't really the best place to be at and that we should try to avoid adding any more of this sort of usage.

Although: The "one mutable XOR any number of immutable" borrow rule is being respected in synchronous ops, right? Since for the duration of the call JS is blocked we're guaranteed to have sole ownership of the buffer. ... Well, except if our call callbacks into JS and the callback mutates the buffer...

Reply:
It's the wild west setting -- there's no guarantees at all!

Reply:
I changed this to `unsafe` for now -- we can revisit